CMakeLists.txt

```diff
@@ -64,31 +60,31 @@ include(check_packages)
 include(swig)
 include(coveralls)
 
-# add PaddlePaddle version
-if(DEFINED ENV{PADDLE_VERSION})
-    add_definitions(-DPADDLE_VERSION=\"$ENV{PADDLE_VERSION}\")
+# Set PaddlePaddle version to Git tag name or Git commit ID.
+find_package(Git REQUIRED)
+execute_process(
+  COMMAND ${GIT_EXECUTABLE} describe --tags --exact-match --abbrev=0
+  WORKING_DIRECTORY ${PROJ_ROOT}
+  OUTPUT_VARIABLE GIT_TAG_NAME
+  RESULT_VARIABLE GIT_RESULT
+  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(NOT ${GIT_RESULT})
+  set(PADDLE_VERSION ${GIT_TAG_NAME})
```

Comment:
I actually tried on my local repo:

```
git tag yi
paddle/scripts/docker/generate.sh
docker build -t paddle:cpu -f paddle/scripts/docker/Dockerfile.cpu .
```

The `docker build` would print the version number it retrieved, and I noticed that it was *yi*.

Is there anything I didn't take enough care of?